core/src/main/scala-3/sttp/tapir/generic/auto/package.scala

```diff
@@ -1,11 +1,15 @@
 package sttp.tapir.generic
 
 import sttp.tapir.Schema
+import sttp.tapir.SchemaType.SString
 import sttp.tapir.internal.SchemaMagnoliaDerivation
+
 import scala.deriving.Mirror
 
 package object auto extends SchemaDerivation
 
 trait SchemaDerivation extends SchemaMagnoliaDerivation {
   inline implicit def schemaForCaseClass[T](implicit m: Mirror.Of[T], cfg: Configuration): Derived[Schema[T]] = Derived(derived[T])
+
+  inline implicit def schemaForEnum[T <: scala.reflect.Enum]: Schema[T] = Schema(SString())
```

Comment:
Isn't `scala.reflect.Enum` in scala3 sth different than `scala.Enumeration`? I think `scala.Enumeration` should work in both scala 2 & scala 3.

`scala.reflect.Enum` should be handled by magnolia, as a sealed case class?

Reply:
right, `scala.Enumeration` is present in scala3

Reply:
in scala 3 there is an extra way to do enums
if you use traditional scala 2 enums you still use the old `scala.Enumeration`,
however new style enums create a sealed class that extends `scala.reflect.Enum` under the covers as per https://docs.scala-lang.org/scala3/reference/enums/enums.html
Based on that as of v1.0.1 using new style enums and the built in derivation will generate a reference to a set of empty objects, rather than anything useful. 

Although that is not that useful, I'm not sure what it should generate by default!  It would depend on the enum.
Perhaps if there are no fields on the sealed class, it could generate a string property whose values based on the name of each enumeration case.
If there are fields on the sealed class, it could generate an object reference where the fields match the ones on the sealed class.

Reply:
@johnduffell Hmm I thought that scala 3 `enum`s should work just fine, and that there is some special casing around case-object-only sealed traits. However if that's not the case, can you create a new issue so that we can investigate?

Reply:
thanks Adam, you are right that they work fine in terms of functionality.  I think I am actually making a feature request to customise the representation.  I have raised https://github.com/softwaremill/tapir/issues/2336 as suggested, with more information so you can understand better.  Hope that is useful.